### PR TITLE
osd: miscellaneous fixes

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
@liewegas Batch of core-related fixes and cleanups. Please help to review, thanks:)

**Comment by xiexingguo** [comment]
@liewegas Set milestone to jewel for attention.  Reset if these changes turn out to be irrelevant.

**Comment by liewegas** [comment]
Several of these I'm nervous about (the locking change, map epoch change, etc).  Let's backport as needed once jewel is out.

File: src/osd/PG.cc [hunk 1]
```diff
@@ -5944,7 +5944,7 @@ boost::statechart::result PG::RecoveryState::Peering::react(const AdvMap& advmap
     post_event(advmap);
     return transit< Reset >();
   }
-  
+  pg->check_full_transition(advmap.lastmap, advmap.osdmap);
```
Comment (liewegas) [inline src/osd/PG.cc]: The Peering and Active states are both sub-states of Started, which means the check should already get handled there, right?

File: src/osd/PG.cc [hunk 2]
```diff
@@ -3569,22 +3569,6 @@ void PG::sub_op_scrub_unreserve(OpRequestRef op)
   clear_scrub_reserved();
 }
 
-void PG::sub_op_scrub_stop(OpRequestRef op)
```
Comment (liewegas) [inline src/osd/PG.cc]: @tchaikov no callers for this now.. but is this something we might want later?  e.g., if 'noscrub' is set on the cluster or something?

File: src/osd/PG.cc (hunk 1 above)
Comment (xiexingguo) [inline src/osd/PG.cc]: Oops, that's true and I am going to drop this change now. And thanks for the review:)

File: src/osd/PG.cc (hunk 2 above)
Comment (tchaikov) [inline src/osd/PG.cc]: @liewegas, sorry for the late response. this change was introduced by @jdurgin in 378f84c1a65875d32ac6b8802a245f1d8de06373, but i don't quite understand how we can stop the ongoing scrub by clearing `scrubber.reserved`.

**Pull request merged**